### ISSUE TITLE
Control Map

### ISSUE THREAD

**Issue by thalesmaoa** [MiG-41/Modbus-GoodWe-DT#1]
Hello, do you have the control address? I need to disable MPPT of my inverter ir control the output power.

**Comment by MiG-41** [comment]
Any control addressees are welcome :)

**Comment by rinivanzetten** [comment]
Hello, Tonight i came across the full specs of the goodwe modbus protocol.
Have not checked it already, but it looks promising.

You can download it at : 
https://www.scribd.com/document/432906859/Modbus-PROTOCOL-Grid-Englishx

You need to subscribe. But first month is free....

**Comment by thalesmaoa** [comment]
After I've got your first answer, I've contacted Goodwe support. However, they've made me sign a NDA. 
I will check how I can contribute without violate NDA.

**Comment by MiG-41** [comment]
@rinivanzetten , Document from ScriBD looks prommising. In free time i'll check this.

@thalesmaoa , i also talked with local GoodWe service. He first said ,that he will check it ,becouse he have acces to GoodWe library , but later he said ,that he is not able to proviede it , since it is not public ,an he will be charged ,if some data will  be published ( so also like NDA).

**Comment by MiG-41** [comment]
I was able to set power factor to -0,8 via command:
`F7  10  01  01  00  01  02  00  14  98  EA`
Values for 7 & 8 bytes are:
00 14 - -0,8
00 13 - -0,81
00 12 - -0,82
00 11 - -0,83
00 10 - 0,84
00 0F - -0.85
00 0A - -0,9
00 01 - -0,99

00 64 - 1

00 63 - +0,99
00 5A - +0,9
00 50 - +0,8

**Comment by thalesmaoa** [comment]
@MiG-41, that worked for me as well. Setting power output is one address bellow, however, I can't that to work.

`F7 10 01 00 00 01 00 HB LB  CRC `

Reconnecting time is 00 01 add.

**Comment by MiG-41** [comment]
I havn't tried , but i belive you miss one byte  ,with is amount of data  and have value 02 :
![image](https://user-images.githubusercontent.com/57936100/102135597-81c2bf00-3e58-11eb-979b-608eb461951e.png)

So it should be something like this :
` F7 10 01 00 00 01 02 00  LB CRC CRC`


So i.e. setting 10% of power i assuming it should be:

` F7 10 01 00 00 01 02 00  0A 19 33`

**Comment by thalesmaoa** [comment]
You are right, but it was only miss typo. I use nodered-modbus and the message seems to be fine since the inverter reply. However, I can't understand why its taking effect for power factor, but not for power limiting. Please, let me know if you try it.

**Comment by MiG-41** [comment]
Problem is ,that when i come back home inverter is already down. Also generally days are foggy and also to see an influance would be hard. 
But you are able to adjust manualy on display menu this power limitation? Can you read value from that adress 0100 each time when you change it in menu ? Then just send this value , maybe it is other logic then normal 1-100%...

**Comment by thalesmaoa** [comment]
I tried. It only return zeros.

**Comment by thalesmaoa** [comment]
I found the problem. HIM does not update but power is limited. Looks like a simple bug.

**Comment by MiG-41** [comment]
So after command inverter limiting power , but read value from 0100 adress not changing ?

**Comment by thalesmaoa** [comment]
It is changing. The value that I write into FC:16, it returns. However, it does not update on screen neither inform that the value is limited by the one shown in screen.
